### PR TITLE
Create Civiform header

### PR DESCRIPTION
**Pull request by greg-hyde**
### Description

Implement a basic header and add it to all existing Thymeleaf pages. It includes the login/logout button and language selector, and uses the built in mobile header

#### General

Read the full guidelines for PRs [here](https://github.com/civiform/civiform/wiki/Technical-contribution-guide#creating-a-pull-request)

- [x] Added the correct label: < feature | enhancement | bug | under-development | dependencies | infrastructure | ignore-for-release | database >
- [x] Assigned to a specific person, `civiform/developers`, or a [more specific round-robin list](https://github.com/civiform/civiform/wiki/Technical-contribution-guide#adding-reviewers)
- [x] Added an additional reviewer from outside your organization as FYI (if the primary reviewer is in the same organization as you)
- [x] Created unit and/or browser tests which fail without the change (if possible)
- [x] Performed manual testing (Chrome and Firefox if it includes front-end changes)

#### User visible changes

- [x] Tested on mobile view. See [mobile device mode](https://developer.chrome.com/docs/devtools/device-mode/)
- [x] Manually tested at 200% size
- [x] Manually evaluated tab order

#### New Features

- [x] Conditioned new functionality on a [FeatureFlag](https://github.com/civiform/civiform/wiki/Feature-Flags)
- [x] Wrote browser tests with the feature flag off and on, etc.

### CODE REVIEW

**Comment by greg-hyde** [comment]
* **#7478** <a href="https://app.graphite.dev/github/pr/civiform/civiform/7478?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @greg-hyde and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

**Review by rockycodes — COMMENTED** [review]
I didn't do a super thorough review as the fyi reviewer, but overall lgtm!

File: server/app/assets/stylesheets/uswds/_uswds-theme-custom-styles.scss [hunk 1]
```diff
@@ -62,3 +62,38 @@ See the radio-option below as an example.
...
+.usa-header--basic .usa-button {
+  @include u-line-height('sans', 2); 
+}
+
+/**
+ * Default form styles add a bottom margin to forms, but we don't want that for the language
+ * selector as it would mess with header styles.
+ */
+
+.usa-header--basic form {
```
Comment (avaleske) [inline server/app/assets/stylesheets/uswds/_uswds-theme-custom-styles.scss:95]: **Question**:  Should we add a specific class on the language selector rather then removing it from all forms in the header? Or will we never want it on any form in the header?

**Review by avaleske — COMMENTED** [review]
Do we maybe want to only take a screenshot of the question itself for most of these, rather then the whole page?

We can have one test that makes sure the question page in general is has the header but otherwise I don't think screenshotting the header on every question permutation is necessary.

File: server/app/assets/stylesheets/uswds/_uswds-theme-custom-styles.scss (hunk 1 above)
Comment (greg-hyde) [reply server/app/assets/stylesheets/uswds/_uswds-theme-custom-styles.scss:95]: If there was another form in the header (which seems unlikely), we would also want to remove that style so it doesn't mess with the alignment of the header buttons

**Comment by greg-hyde** [comment]
> Do we maybe want to only take a screenshot of the question itself for most of these, rather then the whole page?
> 
> We can have one test that makes sure the question page in general is has the header but otherwise I don't think screenshotting the header on every question permutation is necessary.

Yeah agreed. I just filed https://github.com/civiform/civiform/issues/7533 to cover that work

File: server/app/views/applicant/NorthStarApplicantBaseView.java [hunk 2]
```diff
@@ -1,39 +1,125 @@
...
+    context.setVariable(
+        "smallLogoUrl",
+        settingsManifest
+            .getCivicEntitySmallLogoUrl()
+            .orElse(assetsFinder.path("Images/civiform-staging.png")));
+    context.setVariable(
+        "civicEntityShortName", settingsManifest.getWhitelabelCivicEntityShortName(request).get());
+    context.setVariable("closeIcon", Icons.CLOSE);
+
+    // Language selector params
```
[the remark this answers is not in the document]
Comment (avaleske) [reply server/app/views/applicant/NorthStarApplicantBaseView.java:68]: ahhh yeah I guess the dot notation expects it to be a field on a child object, and maybe the underscore notation is more clunky then it's worth. This is where having `record` classes would be nice.

and yeah totally fine in a followup.

**Review by avaleske — DISMISSED** [review]

**Review by avaleske — DISMISSED** [review]

**Review by avaleske — APPROVED** [review]

**Pull request merged**